packages/devtools_app/lib/src/screens/deep_link_validation/deep_link_list_view.dart

```diff
@@ -0,0 +1,356 @@
+// Copyright 2023 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:math';
+
+import 'package:devtools_app_shared/ui.dart';
+import 'package:flutter/material.dart';
+
+import '../../shared/common_widgets.dart';
+import '../../shared/primitives/utils.dart';
+import '../../shared/table/table.dart';
+import '../../shared/table/table_data.dart';
+import '../../shared/ui/colors.dart';
+import '../../shared/utils.dart';
+import 'deep_links_controller.dart';
+import 'deep_links_model.dart';
+
+enum TableViewType {
+  domainView,
+  pathView,
+  singleUrlView,
+}
+
+/// A view that display all deep links for the app.
+class DeepLinkListView extends StatefulWidget {
+  const DeepLinkListView({super.key});
+
+  @override
+  State<DeepLinkListView> createState() => _DeepLinkListViewState();
+}
+
+class _DeepLinkListViewState extends State<DeepLinkListView>
+    with ProvidedControllerMixin<DeepLinksController, DeepLinkListView> {
+  List<String> get androidVariants =>
+      controller.selectedProject.value!.androidVariants;
+
+  @override
+  void initState() {
+    super.initState();
+    callWhenControllerReady((_) {
+      int releaseVariantIndex = controller
+          .selectedProject.value!.androidVariants
+          .indexWhere((variant) => variant.toLowerCase().contains('release'));
+      // If not found, default to 0.
+      releaseVariantIndex = max(releaseVariantIndex, 0);
+      controller.selectedVariantIndex = releaseVariantIndex;
+    });
+  }
+
+  @override
+  void didChangeDependencies() {
+    super.didChangeDependencies();
+    if (!initController()) return;
+  }
+
+  Widget _buildTopPanel() {
+    return Row(
+      mainAxisAlignment: MainAxisAlignment.end,
+      children: [
+        Padding(
+          padding: const EdgeInsets.all(defaultSpacing),
+          child: _buildAndroidVariantDropDown(),
+        ),
+      ],
+    );
+  }
+
+  Widget _buildAndroidVariantDropDown() {
+    final List<DropdownMenuEntry<int>> colorEntries =
+        <DropdownMenuEntry<int>>[];
+    for (int i = 0; i < androidVariants.length; i += 1) {
+      colorEntries.add(
+        DropdownMenuEntry<int>(
+          value: i,
+          label: androidVariants[i],
+        ),
+      );
+    }
+    return DropdownMenu<int>(
+      initialSelection: controller.selectedVariantIndex,
+      label: const Text('Android Variant'),
+      dropdownMenuEntries: colorEntries,
+      onSelected: (int? index) {
+        setState(() {
```

Comment:
I don't think we need this call to setState here since setting `selectedVariantIndex` triggers the linkDatasNotifier to notify. If the set state is necessary to update the dropdown, then maybe `selectedVariantIndex` itself should be a notifier, that way the state can be pushed down the tree to just this dropdown button instead of calling set state on the entire `DeepLinkListView` tree.